src/actions/pause/extra.js

```diff
@@ -0,0 +1,89 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at <http://mozilla.org/MPL/2.0/>. */
+
+// @flow
+
+import { getSymbols, getSource, getSelectedFrame } from "../../selectors";
+import { isReactComponent, isImmutable } from "../../utils/preview";
+import { findClosestClass } from "../../utils/ast";
+
+import type { ThunkArgs } from "../types";
+
+async function getReactProps(evaluate) {
+  const reactDisplayName = await evaluate(
+    "this.hasOwnProperty('_reactInternalFiber') ? " +
+      "this._reactInternalFiber.type.name : " +
+      "this._reactInternalInstance.getName()"
+  );
+
+  return {
+    displayName: reactDisplayName.result
+  };
+}
+
+async function getImmutableProps(expression: string, evaluate) {
+  const immutableEntries = await evaluate((exp => `${exp}.toJS()`)(expression));
+
+  const immutableType = await evaluate(
+    (exp => `${exp}.constructor.name`)(expression)
+  );
+
+  return {
+    type: immutableType.result,
+    entries: immutableEntries.result
+  };
+}
+
+async function getExtraProps(getState, expression, result, evaluate) {
+  const props = {};
+  if (isReactComponent(result)) {
+    const selectedFrame = getSelectedFrame(getState());
+    const source = getSource(getState(), selectedFrame.location.sourceId);
+    const symbols = getSymbols(getState(), source);
+
+    if (symbols && symbols.classes) {
+      const originalClass = findClosestClass(symbols, selectedFrame.location);
+
+      if (originalClass) {
+        props.react = { displayName: originalClass.name };
+      }
+    }
+
+    if (!props.react) {
+      props.react = await getReactProps(evaluate);
```

Comment:
I know originally when I was adding to this function we kept this evaluate function here as a safety net just in case the closest class can't be found, although we thought that going through the symbols should be the standard, as it is, and that this safety-net may not really be needed. Could this be removed, and therefore be able to remove the getReactProps all together?

Reply:
yeah - we'll definitely simplify this